services/state/roster/roster.proto

```diff
@@ -0,0 +1,126 @@
+ * to manage node participation in the Threshold Signature Scheme (TSS).<br/>
+ * All fields are REQUIRED.
+ */
+message RosterEntry {
+
+    /**
+     * A consensus node identifier.
+     * <p>
+     * Node identifiers SHALL be unique _within_ a shard and realm,
+     * but a node SHALL NOT, ever, serve multiple shards or realms,
+     * therefore the node identifier MAY be repeated _between_ shards and realms.
+     */
+    uint64 node_id = 1;
+
+    /**
+     * A consensus weight.
+     * <p>
+     * Each node SHALL have a weight in consensus calculations.<br/>
+     * The consensus weight of a node SHALL be calculated based on the amount
+     * of HBAR staked to that node.<br/>
+     * Consensus SHALL be calculated based on agreement of greater than `2/3`
+     * of the total `weight` value of all nodes on the network.
+     */
+    uint64 weight = 2;
+
+    /**
+     * An RSA public certificate used for signing gossip events.
+     * <p>
+     * This value SHALL be a certificate of a type permitted for gossip
+     * signatures.<br/>
+     * This value SHALL be the DER encoding of the certificate presented.<br/>
+     * This field is REQUIRED and MUST NOT be empty.
+     */
+    bytes gossip_ca_certificate = 3;
+
+    /**
+     * An elliptic curve public encryption key.<br/>
+     * This contains the _long term_ public key for this node.
+     * <p>
+     * This value SHALL be the DER encoding of the presented elliptic curve
+     * public key.<br/>
+     * This field is REQUIRED and MUST NOT be empty.
+     */
+    bytes tss_encryption_key = 4;
+
+    /**
+     * A list of service endpoints for gossip.
+     * <p>
+     * These endpoints SHALL represent the published endpoints to which other
+     * consensus nodes may _gossip_ transactions.<br/>
+     * If the network configuration value `gossipFqdnRestricted` is set, then
+     * all endpoints in this list SHALL supply only IP address.<br/>
+     * If the network configuration value `gossipFqdnRestricted` is _not_ set,
+     * then endpoints in this list MAY supply either IP address or FQDN, but
+     * SHALL NOT supply both values for the same endpoint.<br/>
+     * This list SHALL NOT be empty.<br/>
+     * This list SHALL NOT contain more than `10` entries.<br/>
+     * The first two entries in this list SHALL be the endpoints published to
+     * all consensus nodes.<br/>
+     * All other entries SHALL be reserved for future use.
+     */
+    repeated proto.ServiceEndpoint gossip_endpoint = 5;
+}
+
+/**
+ * A map of roster hashes to roster entries.
+ * The map will be stored in the state as a PBJ map.
+ * It will itself be managed as a Singleton state object.
+ */
+message RosterMap {
+
+    /**
+     * A map of roster hashes to roster entries.
+     * The map will be stored in the state as a PBJ map.
+     * It will itself be managed as a Singleton state object.
+     */
+    map<string, RosterEntry> entries = 1;
+}
```

Comment:
This should probably be removed in favor of the latest design decision to use States API maps for storing the RosterMap.

Reply:
it needs a quick update post roster conversations. Should have that ready by end of day.

Reply:
This RosterMap protobuf model still needs to go.